### PR TITLE
fix(observability): issues with event_cache PR

### PR DESCRIPTION
**Pull request by StephenWakely**
Ref #17580
Ref #17581

Re this PR #17549

This fixes a couple of issues with the mentioned PR.

- When telemetry was turned off it was still emitting the empty tags.
- The service tag was being emitted by converting the value using `to_string`. This meant the value was delimited by `"`. 
- A `telemetry` section is added to the global configuration docs.

### CODE REVIEW

File: website/cue/reference/configuration.cue [hunk 1]
```diff
@@ -252,6 +252,56 @@ configuration: {
...
+										"""
+									required: false
+									type: bool: {
+										default: false
+									}
+								}
+								emit_service: {
+									common: false
+									description: """
+										Emit the service the event was generated by. This is taken from a field of the event
```
Comment (jszwedko) [inline website/cue/reference/configuration.cue:288]:
```suggestion
										Add a `service` tag with the service component the event was received from. If there is
```

For both service and source I think we should also call out how that data is found for logs, metrics, and traces.

File: website/cue/reference/configuration.cue [hunk 2]
```diff
@@ -252,6 +252,56 @@ configuration: {
...
+				Configures options for how Vector emits telemetry.
+				"""
+			required: false
+			type: object: {
+				examples: []
+				options: {
+					tags: {
+						required: false
+						description: """
+							Controls which tags should be emitted with the `vector_component_sent_events_total` and
```
Comment (jszwedko) [inline website/cue/reference/configuration.cue:267]:
```suggestion
							Controls which tags should be included with the `vector_component_sent_events_total` and
```

File: website/cue/reference/configuration.cue [hunk 3]
```diff
@@ -252,6 +252,56 @@ configuration: {
 			}
 		}
 
+		telemetry: {
```
Comment (jszwedko) [inline website/cue/reference/configuration.cue:255]: I take it that this is not generated from the `configurable` trait? Maybe that doesn't work for global options?

Comment (StephenWakely) [reply website/cue/reference/configuration.cue:255]: Yeah I'm pretty sure it only works for components.

Comment (spencergilbert) [reply website/cue/reference/configuration.cue:255]: 👀 but global options are marked with [configurable](https://github.com/vectordotdev/vector/blob/53f8bff371cdfa96770c03be38ae3c83a497043f/lib/vector-core/src/config/global_options.rs#L37)?

Comment (jszwedko) [reply website/cue/reference/configuration.cue:255]: cc/ @tobz to confirm. Maybe it just isn't handled by the Ruby script that turns the JSON schema into cue docs yet.

Comment (StephenWakely) [reply website/cue/reference/configuration.cue:255]: There is no `reference/base` to include them in the cue file from..

**Pull request merged**